jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/BasePathSpec.java

```diff
@@ -0,0 +1,114 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.http.pathmap;
+
+/**
+ * The base PathSpec, what all other path specs are based on
+ */
+public abstract class BasePathSpec implements PathSpec
+{
+    protected static class Decomposition
```

Comment:
I'm not sure I see the benefit of the Decomposition class? What benefit does it give over and above being just  private fields on the BasePathSpec class?

Reply:
OK, I've had an offline conversation with @gregw and I see why you might have introduced the Decomposition class. IMHO the behaviour in the base class is so trivial as to not warrant the complexity of introducing the Decomposition class, so I would ditch the BasePathSpec and have ServletPathSpec, RegexPathSpec directly implement the PathSpec interface.

Reply:
I introduced `Decomposition` more for the bits in common between `RegexPathSpec` and `UriTemplatePathSpec`.

But it's getting apparent now that the code has been written that it's not worth it.